library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -183,20 +183,24 @@ impl Command {
 
         #[cfg(not(target_os = "l4re"))]
         {
+            if let Some(g) = self.get_groups() {
+                //FIXME: Redox kernel does not support setgroups yet
+                #[cfg(not(target_os = "redox"))]
+                cvt(libc::setgroups(g.len().try_into().unwrap(), g.as_ptr()))?;
+            }
             if let Some(u) = self.get_gid() {
                 cvt(libc::setgid(u as gid_t))?;
             }
             if let Some(u) = self.get_uid() {
                 // When dropping privileges from root, the `setgroups` call
-                // will remove any extraneous groups. If we don't call this,
-                // then even though our uid has dropped, we may still have
-                // groups that enable us to do super-user things. This will
-                // fail if we aren't root, so don't bother checking the
-                // return value, this is just done as an optimistic
-                // privilege dropping function.
+                // will remove any extraneous groups. We only drop privileges
+                // if the current uid is 0 and we weren't given an explicit
+                // set of groups.
```

Comment:
Can you keep the part of the comment about being able to do super-user things if this isn't done?

Reply:
You're right. I adjusted the comment to include being able to do super-user things even if the uid is dropped.